app/controllers/articles.js

```diff
@@ -30,116 +31,169 @@ exports.load = function (req, res, next, id){
 exports.index = function (req, res){
   var page = (req.params.page > 0 ? req.params.page : 1) - 1;
   var perPage = 30;
-
   var options = {
-          perPage: perPage,
-          page: page
-        };
+    perPage: perPage,
+    page: page,
+    filters: {}
+  };
+  req.query.title ? options.filters.title = {'$regex': req.query.title} : "";
```

Comment:
As you can see if req.query.title will be present the it will be set in options.filters otherwise options.filters.title will not be added. If req.query.title will not be present then "options.filters.title" will not exist. There will be no undefined.